File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
import java.util.Stack;

class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int i = numberToCheck;
        Stack<Integer> digits = new Stack<>();

        while (true) {
            if (i < 10) {
                digits.add(i);
                break;
            }
            digits.add(i % 10);
            i = i / 10;
        }

        int number_of_digits = digits.size();
        int sum = 0;
        for (Integer digit : digits) {
            sum += java.lang.Math.pow(digit, number_of_digits);
        }

        return sum == numberToCheck;
    }
}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include "math.h"

bool is_armstrong_number(int candidate) {
  int digit_count = _get_digit_count(candidate);
  int n = candidate;
  int sum = 0;

  while (n > 0) {
    int digit = n % 10;
    n = n / 10;
    sum += pow(digit, digit_count);
  }

  return sum == candidate;
}

int _get_digit_count(int n) {
  int c = 0;
  while (n > 0) {
    c++;
    n = n / 10;
  }
  return c;
}
